src/services/walletService.js

```diff
@@ -0,0 +1,43 @@
+import localforage from 'localforage';
+import { BASE_CURRENCY, USERS_PROFILES } from '../utils/constants';
+
+export const addCurrency = (userProfile, currencyName, amount) => {
+  userProfile.funds[currencyName] += amount;
+};
+
+export const getUserById = async (userId) => (
+  (await localforage.getItem(USERS_PROFILES))
+    .find((user) => user.userId === userId)
+);
+
+export default {
```

Comment:
To verify if user exists we may consider https://developer.mozilla.org/pl/docs/Web/JavaScript/Reference/Global_Objects/Proxy but it looks like IE11 doesn't support it, polyfill required.

Reply:
Not sure which trap should I use for object's methods so for now I'm gonna leave it with validator function call which throws error.
I think traps are for object properties and not for methods because we can easily change the behavior of function but not that easily for e.g. getters. Am I correct?